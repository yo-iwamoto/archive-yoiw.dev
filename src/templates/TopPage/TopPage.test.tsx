import { TopPage } from '.';
import { render } from '@testing-library/react';

// Server Component
vi.mock('./components/Blogs', () => ({
  Blogs: () => <div>mock Blogs</div>,
}));

describe('templates/TopPage', () => {
  it('正しく表示されること', () => {
    const { getByRole, asFragment } = render(<TopPage />);

    expect(getByRole('region', { name: 'ブログ' })).toBeDefined();
    expect(getByRole('heading', { name: 'ブログ', level: 2 })).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
