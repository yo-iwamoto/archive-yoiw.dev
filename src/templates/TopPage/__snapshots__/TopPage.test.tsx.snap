// Vitest Snapshot v1

exports[`templates/TopPage > 正しく表示されること 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="blog-heading"
  >
    <h2
      id="blog-heading"
    >
      <span
        class="sr-only"
      >
        ブログ
      </span>
      <span
        aria-hidden="true"
      >
        Blogs
      </span>
    </h2>
    <div>
      mock Blogs
    </div>
  </section>
</DocumentFragment>
`;
